### DISCUSSION TITLE
Trying to set up with for live use with wind controller - Breath control ?

### DISCUSSION THREAD

**Discussion by justinbacle** [probonopd/MiniDexed#522] [Q&A] [answered]
Hello all, and thank you for this impressive project, and especially the nice instructions on how to set everything up!

I am trying to use this project with an AKAI EWI USB, which would be used for live performances.
I managed to get the setup working, using :
- RPi 2B
- 1602 LCD connected in i2c
- PCM5102A audio
- Akai EWI USB
- K4-040 Rotary Encoder
Everything worked plug and play, midi controller is detected and working fine.
I'm quite new to this project, but have been playing with FM synthesis quite a lot a few years back (mostly using, Freq Monster 801 and some Yamaha keyboards)

In order to play correctly sounds with a wind controller, you need to assign the breath control (CC#2 if I'm correct) to the master volume (or individual voices volume).
Can any of you give basic instructions or leads on how to do that?

Thank you :)

**Comment by probonopd** [comment]
> In order to play correctly sounds with a wind controller, you need to assign the breath control (CC#2 if I'm correct) to the master volume (or individual voices volume).

Do you have a pointer (e.g., quote from the manual) on how that works on Yamaha instruments? I don't have a breath controller to work with.

**Reply by Andy2No** [reply]
> > In order to play correctly sounds with a wind controller, you need to assign the breath control (CC#2 if I'm correct) to the master volume (or individual voices volume).
> 
> Do you have a pointer (e.g., quote from the manual) on how that works on Yamaha instruments? I don't have a breath controller to work with.

Here's some snippets from the Yamaha TX7 manual, for reference.  The TX7 is basically a DX7 in sound module form (no built in keyboard), with one major difference - there is a function memory associated with each patch in the user bank, each of which contains settings which are global on the DX7.  Apart from that, it works the same and takes the same patches:


"The TX7 has 32 voice memories, each of which contains the data for one complete 6-operator FM voice. Each of these voice memories also has a function memory which contains the data for the function and control parameters of that voice. The programmable function parameters are given in the following chart:

"
MASTER TUNE
PLAY MODE, POLY/MONO
PITCH WHEEL RANGE
PITCH WHEEL STEP
PORTAMENTO MODE
PORTAMENTO/GLISSANDO SWITCH
PORTAMENTO TIME
MODULATION WHEEL RANGE
MODULATION WHEEL PITCH
MODULATION WHEEL AMPLITUDE
**MODULATION WHEEL EG BIAS**
FOOT CONTROL RANGE
FOOT CONTROL PITCH
FOOT CONTROL AMPLITUDE
**FOOT CONTROL EG BIAS**
BREATH CONTROL RANGE
BREATH CONTROL PITCH
BREATH CONTROL AMPLITUDE
**BREATH CONTROL EG BIAS**
AFTER TOUCH RANGE
AFTER TOUCH PITCH
AFTER TOUCH AMPLITUDE
**AFTER TOUCH EG BIAS**
LIMIT KEY LOWEST
LIMIT KEY HIGHEST
ATTENUATION"

The relevant parts there are the four EG Bias settings, for Mod wheel, foot, breath and aftertouch.

"Real-Time Control Functions

This group of functions pertains to effects which are controlled via controllers built
into or connected to the DX synthesizer being used with the TX7.

Please note that all functions in this group, except the pitch wheel functions, have exactly the same parameters which produce the same effects via the different controllers. The parameters for MODULATION WHEEL, FOOT CONTROLLER, BREATH CONTROLLER and AFTER TOUCH are all the same: depth (0— 15), pitch (off or on), amplitude (off or on) and EG bias (off or on). For these four controllers to function, however, the AMPLITUDE MODULATION SENSITIVITY and PITCH MODULATION SENSITIVITY parameters in the voice data of the cor-responding voice must be set to a value other than 0. The LFO modulation speed setting is also integral with the actual voice data, and must be set as required for
each voice."

"
MODULATION WHEEL EG BIAS
MW = 0,PO, ao,e| 
This "switch" determines whether modulation will be applied to the EG bias of the corresponding voice to produce tonal variations. MODULATION WHEEL EG BIAS Is either 0 (OFF) or 1 (ON)."

"FOOT CONTROL EG BIAS
This "switch" determines whether modulation will be applied to the EG bias of the corresponding voice to produce tonal variations. FOOT CONTROL EG BIAS is either 0 (OFF) or 1 (ON).

BREATH CONTROL RANGE
BC =H,PO, AO, EO
This function determines the maximum depth of modulation which can be applied with the breath controller. The programmable range is to 15, corresponding to the 0 to 99 range provided on the DX7. The desired depth value is selected using the N0/-1 and YES/ + 1 buttons.

BREATH CONTROL PITCH
| BC = 0,P J, AO, EO
This "switch" determines whether modulation will be applied to the pitch of the corresponding voice to produce tremolo or wow type effects. BREATH CONTROL AMPLITUDE is either 0 (OFF) or 1 (ON).

BREATH CONTROL AMPLITUDE
|bC = 0,PQ, aJ, EO
This "switch" determines whether modulation will be applied to the amplitude (volume) of the corresponding voice to produce tremolo or wow type effects. BREATH CONTROL AMPLITUDE is either 0 (OFF) or 1 (ON).

BC = 0,PO,AO,E|
**BREATH CONTROL EG BIAS**
This "switch" determines whether modulation will be applied to the EG bias of the corresponding voice to produce tonal variations. BREATH CONTROL EG BIAS is either 0 (OFF) or 1 (ON).

AFTER TOUCH RANGE
|at = B,PO, AO, EO
This function determines the maximum depth of modulation, as applied with the maximum after touch key pressure. The programmable range is 0 to 15,
corre-sponding to the to 99 range provided on the DX7. The desired depth value is selected using the N0/-1 and YES/ + 1 buttons.

AFTER TOUCH PITCH
AT = 0,P ,A0, EO
This "switch" determines whether modulation will be applied to the pitch of the corresponding voice to produce vibrato type effects. AFTER TOUCH PITCH is either 0 (OFF) or 1 (ON).

AFTER TOUCH AMPLITUDE
AT = 0, PO,Affl,EO 
This "switch" determines whether modulation will be applied to the amplitude (volume) of the corresponding voice to produce tremolo or wow type effects. AFTER TOUCH AMPLITUDE is either 0 (OFF) or 1 (ON).

AFTER TOUCH EG BIAS
AT = 0, PO,AO,E|
This "switch" determines whether modulation will be applied to the EG bias of the corresponding voice to produce tonal variations. AFTER TOUCH EG BIAS is either 0 (OFF) or 1 (ON)."

To recap, the above are the settings for overall control of breath, Mod wheel, aftertouch and foot via EG Bias, which on a DX7 are just global, but can be per user bank patch on a TX7.  There are then sensitivity settings for each operator of a patch, similar to the way operators respond to velocity.

There's at least one family of wind controllers that send Channel Pressure (monophonic Aftertouch) as their breath strength signal - the Casio DH series.  I think the Akai EWI family can all be set up to send that too, and since it's only a two byte message vs three bytes for a CC, that can be a good way to go when there's a high rate of MIDI traffic from the wind controller to the synth module.

The sensitivity settings look like this:
(this is also from the TX7 manual, but quotes the DX7 data format, because it's the same)

"
4-1. DX7 Voice Parameter Change (g = 0)
A. SYSTEM EXCLUSIVE DATA FORMAT

(14) 0P6 AMPLITUDE MODULATION SENSITIVITY 0—3 "

So, each of the six operators can respond to Breath, Aftertouch, MOD or Foot (CC#4), with one of four levels of sensitivity, on the DX7 or TX7.  For TX81Z patches, I think it's 8 levels for each of the four operators.

**Comment by miotislucifugis** [comment]
You can set the Breath Control to "Amplitude"  in the TG's  Modulation menu, no?

**Reply by probonopd** [reply]
Seems about right, looking at https://github.com/probonopd/MiniDexed/wiki/Menu and searching for "breath". But I have no hardware to test it with.

**Reply by miotislucifugis** [reply]
anything that can send CC#2 would work, i recon

**Comment by justinbacle** [comment]
Oh cool, I'll try that this week, thank you !

**Reply by justinbacle** [reply]
Cannot have it to work, even when selecting the "correct" breath modulation to amplitude thing.
One weird thing, is that I tested it in Dexed too, and it doesn't do much either.

I checked the midi data with MidiView, and it shows on channel 1:
`Controller Breath controller (coarse) : XXX`
`Channel pressure XXX`
For each change in breath volume (XXX being the value given, from 0 to 128 it seems)

Midi Hex Codes (for XXX = 125) :
`B0 02 7D`
`D0 7D`

I'd be happy to help debugging and fixing this issue, let me know where to start

**Reply by justinbacle** [reply, marked as answer]
Oh okay, pushing random buttons with dexed, looks like you need to assign breath control to EG Bias (and not amplitude, very counter intuitive, but I guess it's the same on DX7) and then raise A Mod Sens on given operators :o
